neps/search_spaces/hyperparameters/categorical.py

```diff
@@ -39,14 +39,39 @@ def __init__(
         )
         self.value: None | float | int | str = None
 
+        # Check if choices have valid types (float | int | str)
+        for choice in self.choices:
+            if not isinstance(choice, (float, int, str)):
+                raise TypeError(
+                    f'Choice "{choice}" is not of a valid type (float, int, str)')
+
+        # Check if 'default' is in 'choices'
+        if default is not None and default not in self.choices:
+            raise ValueError(
+                f"Default value {default} is not in the provided choices {self.choices}"
+            )
+
+        # Check if 'is_fidelity' is a boolean
+        if not isinstance(is_fidelity, bool):
+            raise TypeError(
+                f"Expected 'is_fidelity' to be a boolean, but got type: "
+                f"{type(is_fidelity).__name__}"
+            )
```

Comment:
Nothing to change, my general opinion is that there's types for a reason and you can't check every input that it's of the correct type. If conflicted on whether to check it or not, I usually favor on not.

However something like `if not 0 <= value <= 1` should be checked as it can't be conveyed in a type.